src/states/Spells.ts

```diff
@@ -151,57 +151,59 @@ export class OvaleSpellsClass implements StateModule {
         targetGUID: string | undefined
     ): [boolean, boolean] {
         this.profiler.StartProfiling("OvaleSpellBook_state_IsUsableSpell");
-        let [isUsable, noMana] = [true, false]
+        let [isUsable, noMana] = [false, false];
         let isKnown = this.OvaleSpellBook.IsKnownSpell(spellId);
-        if (!isKnown) {
-            // check the WOW API if the spell is usable, if it is than we assume that it is a known spell
-            [isUsable, noMana] = IsUsableSpell(spellId); 
-            isKnown = isUsable;
-            this.tracer.Log("Checked WoW API for spell %s. Returns [%s, %s]", spellId, isUsable, noMana);
-        }
         const si = this.ovaleData.spellInfo[spellId];
         if (si && isKnown) {
-            this.tracer.Log(
-                "Found spell info about %s (isKnown = %s)",
-                spellId,
-                isKnown
-            );
-
             const unusable = this.ovaleData.GetSpellInfoProperty(
                 spellId,
                 atTime,
                 "unusable",
                 targetGUID
             );
-            if (unusable !== undefined && unusable > 0) {
+            if (unusable !== undefined && tonumber(unusable) > 0) {
                 this.tracer.Log(
                     "Spell ID '%s' is flagged as unusable.",
                     spellId
                 );
-                isUsable = false;
-            }
-            if (isUsable) {
+                [isUsable, noMana] = [false, false];
+            } else {
                 const seconds = this.TimeToPowerForSpell(
                     spellId,
                     atTime,
                     targetGUID,
                     undefined
                 );
-                noMana = (seconds > 0);
-                if (noMana) {
-                    isUsable = false;
+                if (seconds > 0) {
                     this.tracer.Log(
                         "Spell ID '%s' does not have enough power.",
                         spellId
                     );
+                    [isUsable, noMana] = [false, true];
                 } else {
                     this.tracer.Log(
-                        "Spell ID '%s' passed power requirements.",
+                        "Spell ID '%s' meets power requirements.",
                         spellId
                     );
+                    [isUsable, noMana] = [true, false];
                 }
             }
-        } 
+        } else {
+            const [index, bookType] = this.OvaleSpellBook.GetSpellBookIndex(
+                spellId
+            );
+            if (index && bookType) {
+                [isUsable, noMana] = IsUsableSpell(index, bookType);
+            } else if (isKnown) {
+                const name = this.OvaleSpellBook.GetSpellName(spellId);
```

Comment:
I think that this code can be simplified, as IsUsableSpell works with a spell id, now. I think that if the goal is to work with a spell with the same name but a different id, it should not be done here.

Reply:
Yes, I think that bit was originally because the WoW API function `IsUsableSpell()` only took either `(index, bookType)` or `name` as parameters, but that whole bit can be simplified now that it can take a spell ID instead.  Arguably we can simplify `states/SpellBook.ts` since we no longer need to care about `(index, bookType)`.

Reply:
I'll change this bit in the pullup to just directly call `IsUsableSpell()` with the spell ID instead of the `(index, bookType)` dance here, and I'll make a separate, larger change to `states/SpellBook.ts` to remove the whole `(index, bookType)` tracking from there as well since it's not needed anymore to identify a player's spells.